src/plugins/charts/public/static/components/index.ts

```diff
@@ -0,0 +1,31 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+export { BasicOptions } from './basic_options';
+export { ColorModes, Rotates } from './collections';
+export { ColorRanges, SetColorRangeValue } from './color_ranges';
+export { ColorSchemaOptions, SetColorSchemaOptionsValue } from './color_schema';
+export { ColorSchemaVislibParams, Labels, Style } from './types';
+export { NumberInputOption } from './number_input';
+export { RangeOption } from './range';
+export { RequiredNumberInputOption } from './required_number_input';
+export { SelectOption } from './select';
+export { SwitchOption } from './switch';
+export { TextInputOption } from './text_input';
+export { useValidation } from './utils';
```

Comment:
@sulemanof I noticed that there are two version of `useValidation`: in [vis_default_editor](https://github.com/elastic/kibana/blob/master/src/plugins/vis_default_editor/public/components/controls/utils/agg_utils.ts#L102) and in [vis_type_vislib](https://github.com/elastic/kibana/blob/master/src/legacy/core_plugins/vis_type_vislib/public/components/common/utils.ts#L22). I believe it makes sense to leave just one version (I guess from `vis_default_editor`). What do you think?

Reply:
These are different, check out interfaces there.
So the first one respects particular `paramName` to be invalid

Reply:
I mean to use the version from `vis_default_editor` and a bit change `useValidation` usage in [`RequiredNumberInputOption`](https://github.com/elastic/kibana/blob/0b50ca874e7597434de865295916a4486f5c57a9/src/plugins/charts/public/static/components/required_number_input.tsx#L60):
```tsx
  const setControlValidity = useCallback((isValidControl: boolean) => {
    setValidity(paramName, isValidControl);
  }, [setValidity]);
  useValidation(setControlValidity, isValid);
```

Also the `RequiredNumberInputOption` control is currently used in [`vis_type_vislib`](https://github.com/elastic/kibana/blob/master/src/legacy/core_plugins/vis_type_vislib/public/components/options/point_series/threshold_panel.tsx#L25) only. So I think we don't need to move it to `charts` plugin.

Reply:
I agree it's confusing to have to slightly different helpers with different names. Let's consolidate into one if that's possible without major changes.

About `RequiredNumberInputOption`: OTOH it might be helpful somewhere else too later, not really a preference here.

Reply:
Makes sense.
If it is the only usage, I would also keep the only `useEffect` here.
But I'm fine with both.

Reply:
> About `RequiredNumberInputOption`: OTOH it might be helpful somewhere else too later, not really a preference here.

Yep, there is also an [issue](https://github.com/elastic/kibana/issues/49611) created for this purpose